students/aravichander/lesson07/Mailroom_unittest.py

```diff
@@ -0,0 +1,33 @@
+import unittest
+
+from Mailroom_lesson07 import createlist,getlist,sumdonor
+
+class MyFuncTestCase(unittest.TestCase):
+	
+	def test_createlist(self):
+		createlist()
+		output = getlist()
+		self.assertEqual(output["Bill Gates"],[10000,12000]) 
+		self.assertTrue(output["Bill Gates"] == [10000,12000]) 
+		self.assertFalse(output["Bill Gates"] == [10000,12001])
+		with self.assertRaises(KeyError):
+			output["Carrot Top"]
+
+#This was a failed testing scenario as the original function already handles the exception
```

Comment:
so this is a good hint that you want to refactor your code :-)

you want a function (or method, when you go OO) that finds a given donor -- 

What should it do if it finds the donor? probably return the donor dat structure or object.

What should it do if the donor is not there? It should either return a "null" value -- i.e. None, or raise an exception. Once you decide that, then you know what to test for.